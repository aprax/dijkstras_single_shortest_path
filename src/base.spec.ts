import findShortestPath from ".";
import type { Nodes } from './types';

describe('should return base result if root node ', () => {
	const nodes: Nodes = new Map([
		['A', [['B', 2], ['C', 3]]],
		['B', []],
	]);

	it('does not exist', () => {
		expect(findShortestPath('C', nodes)).toMatchSnapshot();
	})

	it('has no edges', () => {
		expect(findShortestPath('B', nodes)).toMatchSnapshot();
	})
});
