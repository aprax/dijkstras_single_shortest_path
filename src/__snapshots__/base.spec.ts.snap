// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return base result if root node  does not exist 1`] = `
Map {
  "C" => Array [
    "C",
    0,
  ],
}
`;

exports[`should return base result if root node  has no edges 1`] = `
Map {
  "B" => Array [
    "B",
    0,
  ],
}
`;
